jbmc/unit/java_bytecode/java_replace_nondet/replace_nondet.cpp

```diff
@@ -110,26 +146,50 @@ void load_and_test_method(
   remove_virtual_functions(model_function);
 
   // Then test both situations.
+  THEN("Replace nondet should work after remove returns has been called.")
+  {
+    remove_returns(model_function, [](const irep_idt &) { return false; });
+
+    replace_java_nondet(model_function);
+
+    validate_nondet_method_removed(goto_function.body.instructions);
+  }
+
+  THEN("Replace nondet should work before remove returns has been called.")
+  {
+    replace_java_nondet(model_function);
+
+    remove_returns(model_function, [](const irep_idt &) { return false; });
+
+    validate_nondet_method_removed(goto_function.body.instructions);
+  }
+
+  object_factory_parameterst params{};
+
   THEN(
-    "Code should work when remove returns is called before "
-    "replace_java_nondet.")
+    "Replace and convert nondet should work after remove returns has been "
```

Comment:
Bit concerned that we never test that `validate_nondets_converted` fails if we don't call `convert_nondet` at all. That would require restructuring it to return `nondet_exists` though.